src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -0,0 +1,391 @@
+	LUNAR_ETO_ROKUYOU = 19,     // reserved to match Office but not implemented in our code
+	KOREANLUNISOLAR = 20,
+	TAIWANLUNISOLAR = 21,
+	PERSIAN = 22,
+	UMALQURA = 23,
+	LAST_CALENDAR = 23      // Last calendar ID
+};
+
+/*
+* These values should be kept in sync with System.Globalization.CalendarDataType
+*/
+enum CalenderDataType : int32_t
+{
+	Uninitialized = 0,
+	NativeName = 1,
+	MonthDay = 2,
+	ShortDates = 3,
+	LongDates = 4,
+	YearMonths = 5,
+	DayNames = 6,
+	AbbrevDayNames = 7,
+	MonthNames = 8,
+	AbbrevMonthNames = 9,
+	SuperShortDayNames = 10,
+	MonthGenitiveNames = 11,
+	AbbrevMonthGenitiveNames = 12,
+	EraNames = 13,
+	AbbrevEraNames = 14,
+};
+
+// the function pointer definition for the callback used in EnumCalendarInfo
+typedef bool(*EnumCalendarInfoCallback)(const UChar*, void*);
+
+/*
+Function:
+GetCalendarName
+
+Gets the associated ICU calendar name for the CalendarId.
+*/
+const char* GetCalendarName(CalendarId calendarId)
+{
+	switch (calendarId)
+	{
+	case JAPAN:
+		return "japanese";
+	case THAI:
+		return "buddhist";
+	case HEBREW:
+		return "hebrew";
+	case CHINESELUNISOLAR:
+	case KOREANLUNISOLAR:
+	case JAPANESELUNISOLAR:
+	case TAIWANLUNISOLAR:
+		return "chinese";
+	case PERSIAN:
+		return "persian";
+	case HIJRI:
+	case UMALQURA:
+		return "islamic";
+	case GREGORIAN:
+	case GREGORIAN_US:
+	case GREGORIAN_ARABIC:
+	case GREGORIAN_ME_FRENCH:
+	case GREGORIAN_XLIT_ENGLISH:
+	case GREGORIAN_XLIT_FRENCH:
+	case KOREA:
+	case JULIAN:
+	case LUNAR_ETO_CHN:
+	case LUNAR_ETO_KOR:
+	case LUNAR_ETO_ROKUYOU:
+	case SAKA:
+	case TAIWAN:
+	default:
+		return "gregorian";
+	}
+}
+
+/*
+Function:
+GetMonthDayPattern
+
+Gets the Month-Day DateTime pattern for the specified locale.
+*/
+bool GetMonthDayPattern(Locale& locale, UChar* sMonthDay, int32_t stringCapacity)
+{
+	UErrorCode err = U_ZERO_ERROR;
+	LocalPointer<DateTimePatternGenerator> generator(DateTimePatternGenerator::createInstance(locale, err));
+	UnicodeString monthDayPattern = generator->getBestPattern(UnicodeString("MMMMd"), err);
+
+	if (!U_SUCCESS(err))
+		return false;
+
+	monthDayPattern.extract(sMonthDay, stringCapacity, err);
+
+	return U_SUCCESS(err);
+}
+
+/*
+Function:
+GetNativeCalendarName
+
+Gets the native calendar name.
+*/
+bool GetNativeCalendarName(Locale& locale, CalendarId calendarId, UChar* nativeName, int32_t stringCapacity)
+{
+	LocalPointer<LocaleDisplayNames> displayNames(LocaleDisplayNames::createInstance(locale));
+
+	UnicodeString calendarName;
+	displayNames->keyValueDisplayName("calendar", GetCalendarName(calendarId), calendarName);
+
+	UErrorCode err = U_ZERO_ERROR;
+	calendarName.extract(nativeName, stringCapacity, err);
+
+	return U_SUCCESS(err);
+}
+
+/*
+Function:
+GetCalendarInfo
+
+Gets a single string of calendar information by filling the result parameter with the requested value.
+*/
+extern "C" bool GetCalendarInfo(UChar* localeName, CalendarId calendarId, CalenderDataType dataType, UChar* result, int32_t resultCapacity)
```

Comment:
Please use int32_t instead of bool on methods exposed to native code.  Then use a managed bool type on the managed side with a [MarshalAs(UnmanagedType.Bool)] attribute.  We want to be on the Win32 4-Byte Boolean plan.